### DISCUSSION TITLE
v0.10.0

### DISCUSSION THREAD

**Discussion by banesullivan** [banesullivan/localtileserver#184] [Announcements]
## What's Changed
* Release 0.10.0: migration to rio-tiler by @banesullivan in https://github.com/banesullivan/localtileserver/pull/183

This release marks a significant re-implementation of localtileserver. The backing tiler has been changed from large-image to [rio-tiler](https://github.com/cogeotiff/rio-tiler). The overall result is improved performance, simplified dependencies, improved compatibility with the rasterio ecosystem, and many bug/quirk fixes like #172, #159, #147

This release unfortunately has many breaking changes and some loss of features. The breaking changes are mostly around API renaming and some methods changing to properties. Otherwise, I stripped out features that are just better handled upstream or not needed here.

This may be a painful upgrade considering how much has changed. Highlights:

- `band` is now `indexes` to follow rio-tiler's implementation (deprecation warning is issued in parts of the API)
- `cmap` is now `colormap` (deprecation warning is issued in parts of the API)
- `palette` is now `colormap`
- Custom colormaps are no longer supported
- No more region of interest extraction
- Access to underlying `rasterio` dataset is now via the `.dataset` property
- COG validation no longer raises errors, but returns a True/False
- `metadata()` method is now a `metadata` property
- No more support for remote servers (niche feature)
- Some return values of things like bounds, metadata, etc may have changed now that they are fetched from rio-tiler

For those looking for more details, I recommend checking out the diff specifically for the test directory to see what has changed in #183 

<hr /><em>This discussion was created from the release <a href='https://github.com/banesullivan/localtileserver/releases/tag/v0.10.0rc2'>v0.10.0rc2</a>.</em>

**Comment by giswqs** [comment]
Excited to see all these improvements! The support for custom colormaps will be missed.

**Reply by banesullivan** [reply]
> The support for custom colormaps will be missed.

Yeah, hopefully using named matplotlib ones will be okay in the meantime. We can re-add this functionality pretty easily I think, I just need a few examples of how people would like to do it. Would you be able to open a new issue with an example or two of custom colormaps?

**Reply by giswqs** [reply]
Sure, will do.

**Comment by banesullivan** [comment]
cc @vincentsarago, you might be pleased to see I made the switch to rio-tiler. I'd love to get your take on this and my usage of rio-tiler in `localtileserver/tiler/handler.py`. 

Further, I'm curious how you see localtileserver and [rio-viz](https://github.com/developmentseed/rio-viz) coexisting in the Jupyter world? My impression is rio-viz is mostly targeted as a standalone web app that could be used in Jupyter, is that right?

**Reply by vincentsarago** [reply]
👍 looks good 👏 

> My impression is rio-viz is mostly targeted as a standalone web app that could be used in Jupyter, is that right?

Absolutely, rio-viz audience wasn't targeted at Jupyter at first. I'm still not really please with the Jupyter integration anyway so I might get rid of it at one point!
